crates/vm/src/vm.rs

```diff
@@ -154,6 +169,15 @@ impl Vm {
             match update {
                 Some(ProgramControlFlow::Pc(new_pc)) => self.pc = new_pc,
                 Some(ProgramControlFlow::Halt) => break,
```

Comment:
One thing that isn't handled is halting on a child. That should exit the entire program but would only exit the child. This could be a follow up though.